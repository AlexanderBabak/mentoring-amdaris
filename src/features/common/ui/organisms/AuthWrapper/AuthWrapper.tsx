import { FC, PropsWithChildren } from "react";
import { Box, Stack } from "@mui/material";

const AuthWrapper: FC<PropsWithChildren> = ({ children }) => (
  <Stack
    sx={{
      bgcolor: (theme) => theme.palette.primary.contrastText,
      flexGrow: 1,
      justifyContent: "center",
      placeItems: "center",
    }}
  >
    <Box
      sx={{
        bgcolor: (theme) => theme.palette.background.blue,
        borderRadius: 2,
        border: "1px solid",
        borderColor: (theme) => theme.palette.primary.main,
        p: 6,
      }}
    >
      {children}
    </Box>
  </Stack>
);

export default AuthWrapper;
