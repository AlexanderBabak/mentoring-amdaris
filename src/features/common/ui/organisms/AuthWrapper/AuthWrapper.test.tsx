import { render } from "@testing-library/react";
import AuthWrapper from "./AuthWrapper";

describe("MainWrapper", () => {
  it("should work as expected", async () => {
    const { baseElement } = render(
      <AuthWrapper>
        <div data-testid="mock-Component" />
      </AuthWrapper>,
    );

    expect(baseElement).toMatchSnapshot();
  });
  it("reflect background color by bgcolor props", async () => {
    const { baseElement } = render(
      <AuthWrapper>
        <div data-testid="mock-Component" />
      </AuthWrapper>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
