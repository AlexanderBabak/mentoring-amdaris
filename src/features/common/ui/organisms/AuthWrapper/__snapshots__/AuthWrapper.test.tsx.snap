// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainWrapper reflect background color by bgcolor props 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-image: linear-gradient(to bottom, #1C2244, #009DC5, #1C2244);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  place-items: center;
}

.emotion-1 {
  border-radius: 8px;
  border: 1px solid;
  border-color: #1976d2;
  padding: 48px;
}

<body>
  <div>
    <div
      class="MuiStack-root emotion-0"
    >
      <div
        class="MuiBox-root emotion-1"
      >
        <div
          data-testid="mock-Component"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`MainWrapper should work as expected 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-image: linear-gradient(to bottom, #1C2244, #009DC5, #1C2244);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  place-items: center;
}

.emotion-1 {
  border-radius: 8px;
  border: 1px solid;
  border-color: #1976d2;
  padding: 48px;
}

<body>
  <div>
    <div
      class="MuiStack-root emotion-0"
    >
      <div
        class="MuiBox-root emotion-1"
      >
        <div
          data-testid="mock-Component"
        />
      </div>
    </div>
  </div>
</body>
`;
